services-core/src/economic_viability.rs

```diff
@@ -250,4 +282,52 @@ mod tests {
             U256::from(5787037037037u128)
         );
     }
+
+    #[test]
+    fn combined_strategy_picks_min_min_average_fee() {
+        for (fixed_fee, dynamic_fee, expected_fee) in &[(5, 10, 5), (5, 1, 1)] {
+            let fixed = FixedEconomicViabilityComputer::new(*fixed_fee, 0.into());
+            let mut dynamic = MockEconomicViabilityComputing::new();
+            dynamic
+                .expect_min_average_fee()
+                .times(1)
+                .returning(move || Ok(*dynamic_fee));
+            let combined = CombinedEconomicViabilityComputer {
+                fixed,
+                dynamic: Box::new(dynamic),
+            };
+            let result = combined.min_average_fee().now_or_never().unwrap().unwrap();
+            assert_eq!(result, *expected_fee);
+        }
+    }
+
+    #[test]
+    fn combined_strategy_picks_correct_max_gas_price() {
+        let fixed_gas = U256::from(1);
+        let fixed_fee = 15u128;
+        let dynamic_gas = U256::from(2);
+        // In the first run the average fee of the solution is >= the fixed_fee so the fixed gas
+        // price should be chosen and in the second run < so the dynamic should be chosen.
+        for (earned_fee, expected_gas) in &[(15, 1), (14, 2)] {
+            let fixed = FixedEconomicViabilityComputer::new(fixed_fee, fixed_gas);
```

Comment:
ubernit: Everything until info could go outside the loop to make the distinction of the two test cases shorter. 

I'd even prefer two separate tests `combined_strategy_picks_fixed_gas_price_if_fee_greater_fixed_average_fee` and `combined_strategy_picks_dynamic_gas_price_if_fee_smaller_fixed_average_fee` (which would avoid the vague word "correct" in the test description). but I remember you don't like duplicated code and it's not a big deal.

Reply:
Moved the code outside of the loop.